handlers_global.go

```diff
@@ -10,26 +11,26 @@ import (
 	"time"
 
 	"github.com/stripe/veneur/samplers"
-
-	"golang.org/x/net/context"
+	"github.com/stripe/veneur/ssf"
+	"github.com/stripe/veneur/trace"
 )
 
 type contextHandler func(c context.Context, w http.ResponseWriter, r *http.Request)
 
 func (ch contextHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	// nil context is bad form, but since we don't use it, it's a quick hack
-	// to allow us to write tests before we upgrade to Go 1.7 Context type
-	// and the newer goji
-	// TODO(aditya) actually update this
-	ch(nil, w, r)
+	ctx := context.Background()
```

Comment:
From @cory-stripe: `those are usually implemented as headers, fwict` (ie, the IDs are passed as headers).


So the client would initiate a trace and create a span (or more than one) on its flush. It would pass the trace ID in the request as a header to the server, which then uses that trace ID as the root for its traces.